### ISSUE TITLE
[TASK] TOREX Contract

### ISSUE THREAD

**Issue by tokdaniel** [superfluid-finance/superfluid-stories#537]
## Overview
The Torex Contract is a SuperApp providing an asset pair such as USDCx -> ETHx, and is responsible for receiving and handling incoming streams, and providing funcionality to settle trades.

**What?**
Contract implementing stream-in token A -> api to distribute-out token B

**How?**

**Why?**

## Acceptance criteria
 - [ ] supports a pair of tokens (unidirectional)
 - [ ] can receive a stream
 - [ ] can deal with updating a stream
 - [ ] can deal with deleting a stream
 - [ ] notifies AvereageX on an incoming stream, to distribute rewards.
 - [ ] provides a way for users to recover unsettled funds

## Readiness criteria
- [x] Descriptive title
- [x] What?
- [ ] Why?
- [ ] How?
- [x] Acceptance criteria complying with Definition of Done
- [x] Estimate/Story points
- [x] Deadline: If not a part of an epic
- [x] Priority: Reflected as order on the board, walking the board from the top and right
- [x] Assignee
- [x] Labels
- [x] Link to epic: If belonging to an epic
- [x] A task is always a separate ticket/issue
- [x] UI/UX Design (optional)
- [x] Blockers/Prerequisites (optional): Use Zenhub blocked by/blocking feature
